### PR TITLE
Feature/rest api

### PR DESCRIPTION
**Pull request by lucatume**
Tickets:

* https://central.tri.be/issues/70395
* https://central.tri.be/issues/70530
* https://central.tri.be/issues/70711
* https://central.tri.be/issues/71337

This PR adds:

* generic REST API code support; the idea being we could build more and need common stuff (not specifically just events)
* shiny new test facilities

### CODE REVIEW

**Comment by bordoni** [comment]
Would be hella nice to have @borkweb and @barryhughes to chime in here and Code Review this puppy.

**Comment by lucatume** [comment]
Would be hella nice to have everyone!

File: tests/_support/ea-server-mocker/js/ea-mocker.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+(function ( $, undefined ) {
+	var formatJSONString = function ( text ) {
```
Comment (borkweb) [inline tests/_support/ea-server-mocker/js/ea-mocker.js]: My only nitpick in this file is that we use snake_case for our JS vars and function names.

Comment (borkweb) [inline tests/_support/ea-server-mocker/js/ea-mocker.js]: Since this is an internal tool, I'm less concerned about it, though.

**Comment by borkweb** [comment]
Whoa - nice work in here. I :heart: the mocker!

**Comment by borkweb** [comment]
I had added *merge* and then reviewed it so that @barryhughes can have a chance to look it over.

File: tests/_support/ea-server-mocker/js/ea-mocker.js (hunk 1 above)
Comment (lucatume) [inline tests/_support/ea-server-mocker/js/ea-mocker.js]: Good to know; [WP Coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/) go for snakeCase in JS so I stuck to that.
Coding standards are my black beast.

**Comment by lucatume** [comment]
I've adde the "on hold" label to remind me Barry needs to take a look at this and I should not merge it.

**Comment by lucatume** [comment]
This code is ready for code review as the Other URL / REST API functionality is ready enough for QA

**Comment by borkweb** [comment]
_Excellent_ work! 🔢

**Pull request merged**